assignment4/assignment4.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <math.h>
+#include <vector>
+
+class IslandCounter{
+	void collapse_island(std::vector< std::vector<bool> > &map, int i, int j){
+		if(!map[i][j]) return;
+
+		map[i][j] = false;
+		collapse_island( map, std::min(i+1, (int) (map.size()-1)), j);
+		collapse_island( map, std::max(i-1, 0), j);
+		collapse_island( map, i, std::min(j+1,(int) (map[i].size()-1)));
+		collapse_island( map, i, std::max(j-1,0));
+		
+		return;
+	}
+public:
+	int Count(std::vector< std::vector<bool> > map){
```

Comment:
It is still necessary to copy the map once, but I agree it looks cleaner. It should be better for recursion also.